32 bit compiles fail

### Version of Apptainer

1.1.0-rc.1

Attempted 32 bit compiles for fedora fail with the error:

```
# github.com/apptainer/apptainer/internal/pkg/util/fs/overlay
../internal/pkg/util/fs/overlay/overlay_linux.go:84:18: invalid operation: stfs.Type == allowType (mismatched types int32 and int64)
make: *** [Makefile:355: cmd/starter/c/starter] Error 2
```

--- a/internal/pkg/util/fs/overlay/overlay_linux.go
+++ b/internal/pkg/util/fs/overlay/overlay_linux.go
@@ -81,7 +81,7 @@ func check(path string, d dir, allowType int64) error {
 		return nil
 	}
 
-	if stfs.Type == allowType {
+	if int64(stfs.Type) == allowType {
 		return nil
 	}
 

